tests/integration/s2n_handshake_test_gnutls.py

```diff
@@ -28,16 +28,24 @@
 from multiprocessing.pool import ThreadPool
 from s2n_test_constants import *
 
-def try_gnutls_handshake(endpoint, port, priority_str):
+def try_gnutls_handshake(endpoint, port, priority_str, mfl_extension_test):
     # Fire up s2nd
-    s2nd = subprocess.Popen(["../../bin/s2nd", "-c", "test_all", str(endpoint), str(port)], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    if mfl_extension_test:
```

Comment:
I think it's cleaner if we refactor this to use conditionals that build the command list. Then we can call `subprocess.Popen` once at the end. See https://github.com/awslabs/s2n/blob/master/tests/integration/s2n_handshake_test_s_client.py#L77 for an example more clear than my words. What do you think?